### ISSUE TITLE
Improve accessing assets from different packs (main, audio, speech)

### ISSUE THREAD

**Issue by ghost** [adventuregamestudio/ags#963]
The existing engine's mechanism for getting assets when they are in different packs is awful: it requires to reset stored information about previous package's table of contents and read new one's TOC; and it does so *every time* engine has to play a sound or voice-over clip (I think the only exception is when sound cache is used).

This must be changed. There are probably two common solutions:
* keep multiple separate asset libraries loaded and then either pass package name along with every asset name, or search all of them one by one;
* have one asset library composed of multiple packs; in this case asset information should also include original pack's name, or some kind of a pack id (and have pack names in a table where that id serves an index).<br>
**EDIT:** actually, in the latter case the asset request may omit package name at all. Maybe it's the better way: just register all available packs at once in a single TOC and allow for an asset to be present anywhere.

**EDIT:** There's also a minor issue of engine searching up for the audio/speech pack file every time.  This is related to "run from IDE" mode. You may find this by looking for `find_assetlib`. Probably this may be optimized by remembering these locations at startup and reusing them later.

**Comment by rofl0r** [comment]
alternative is to unpack/repack with agsutils and have everything in a single packfile.
i'd assume the split archives func was added to fit a game onto multiple floppies...

**Comment by ivan-mogilko** [comment]
I don't think it's proper to rely on users repacking the games, therefore this does not sound as an alternative, but rather a workaround (and engine would have to be changed anyway, since currently it looks in the hardcoded package names).

Having separate packages could be pretty beneficial, e.g. in case you want to be able to send game without data which may be safely omitted from the game (like digital speech).

**Comment by ivan-mogilko** [comment]
Resolved, relevant commits: c427ee98b1333cc0fd5be0e51e672e0d2e6ca6dc, 3b22ee9f37ba4e052bc96bba4ad03c50cb67d2c5, b0da31a704e37c6176dc7c45fd456c9cb10f36b2

What has changed:
* AssetManager now allows to register multiple asset locations;
* requires to register directories explicitly, no longer does implicit searches in the "current working dir" (one may add current working dir, or relative path to manager, but it will be resolved into absolute path);
* allows to assign filter tags to locations, and use a filter hint when searching for an asset, narrowing down search to only certain paths.
* 
The latter is a workaround to current engine rule where some assets are looked only in particular package. This may or may not be important to maintain, but I decided to replicate this behavior for now, as AGS currently does not have a proper tree-like asset structure (all assets are, so to speak, in the root namespace), so there's no better way to distinct search categories.
This is easy to disable too, as these filters are only internal and do not add anything to data format.
